src/strife/r_data.c

```diff
@@ -690,38 +691,124 @@ void R_InitSpriteLumps (void)
 //
 void R_InitColormaps (void)
 {
+#ifndef CRISPY_TRUECOLOR
     int	lump;
 
     // Load in the light tables, 256 byte align tables.
     lump = W_GetNumForName(DEH_String("COLORMAP"));
     colormaps = W_CacheLumpNum(lump, PU_STATIC);
-    pal_color = colormaps;
+#else
+	int c, i, j = 0;
+	byte r, g, b;
 
-    // [crispy] initialize color translation and color strings tables
-    {
-        byte *playpal = W_CacheLumpName("PLAYPAL", PU_STATIC);
-        char c[3];
-        int i, j;
+	byte *const playpal = W_CacheLumpName("PLAYPAL", PU_STATIC);
+	byte *const colormap = W_CacheLumpName("COLORMAP", PU_STATIC);
 
-        if (!crstr)
-            crstr = I_Realloc(NULL, CRMAX * sizeof(*crstr));
+	if (!colormaps)
+	{
+		colormaps = (lighttable_t*) Z_Malloc((NUMCOLORMAPS + 1) * 256 * sizeof(lighttable_t), PU_STATIC, 0);
+	}
 
-        // [crispy] CRMAX - 2: don't override the original GREN and BLUE2 Boom tables
-        for (i = 0; i < CRMAX - 2; i++)
-        {
-            for (j = 0; j < 256; j++)
-            {
-                cr[i][j] = V_Colorize(playpal, i, j, false);
-            }
+	if (crispy->truecolor)
+	{
+		for (c = 0; c < NUMCOLORMAPS; c++)
+		{
+			const float scale = 1. * c / NUMCOLORMAPS;
+
+			for (i = 0; i < 256; i++)
+			{
+				const byte k = colormap[i];
+
+				if (i < 224)
+				{
+					r = gamma2table[crispy->gamma][playpal[3 * k + 0]] * (1. - scale) + gamma2table[crispy->gamma][0] * scale;
+					g = gamma2table[crispy->gamma][playpal[3 * k + 1]] * (1. - scale) + gamma2table[crispy->gamma][0] * scale;
+					b = gamma2table[crispy->gamma][playpal[3 * k + 2]] * (1. - scale) + gamma2table[crispy->gamma][0] * scale;
+				}
+				else
+				{
+					// [JN] Vanilla Strife is using COLORMAP indexes 224-255 without fading to black
+					// so they work as brightmaps. To replicate it, lock such indexes on first light level.
+					r = gamma2table[crispy->gamma][playpal[3 * k + 0]] + gamma2table[crispy->gamma][0] * scale;
```

Comment:
Well, _technically_ it's always "palette indexes", but _practically_, in this case it's a `COLORMAP` indexes. Have a look:
![image](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/bc19b3ec-80e7-4e11-965e-5766127f37a7)

Yellow and next colors are never diminishing, which works as brightmaps in Strife. Blending is still needed, since colors must be affected by gamma-correction, and `colormaps` array have to be properly filled. Or am I getting something wrong and there is a more simple way to keep these colors always bright?

Reply:
I think the second part of the additions in the formulas must go. The code should look equal to the one in the `pal_color[]` calculation, without the `& 3` part.

Reply:
That's worked (https://github.com/fabiangreffrath/crispy-doom/pull/1183/commits/a711fe8dc6e583eaa18367ab68cdec4630b81223), thanks!